app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Any, Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_box = {}
+
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        key = (args, frozenset(kwargs.items()))
+        if key in cache_box:
+            print("Getting from cache")
+            return cache_box[key]
+        else:
+            print("Calculating new result")
```

Comment:
Code Cleanliness issue: Same as the previous comment, this print statement seems to be for debugging purposes and can be removed from the final version of the code.